### DISCUSSION TITLE
How can I disable the return line wrapping?

### DISCUSSION THREAD

**Discussion by microwin7** [redhat-developer/vscode-java#2395] [Q&A]
![image](https://user-images.githubusercontent.com/12544425/162121040-ec4a767d-8fd5-44b8-9c0f-1c78b80ffd3b.png)

The fact is that I am working on a project with other developers, but their IDEA, where there is no such if return formatting, I tried switching java.format.onType.enabled and other different settings, but nothing helps me.

![image](https://user-images.githubusercontent.com/12544425/162121453-3dc726d9-e5a4-465c-aff9-ec75c01813ef.png)

My changes end up looking ridiculous if I do a Git Push

**Comment by snjeza** [comment]
@microwin7 you may want to take a look at https://github.com/redhat-developer/vscode-java/wiki/Formatter-settings
You can check the following property:
```
org.eclipse.jdt.core.formatter.keep_then_statement_on_same_line=true
```

**Reply by microwin7** [reply]
I immediately changed it myself, that’s what helped me, I didn’t figure out what their differences are
```
org.eclipse.jdt.core.formatter.keep_imple_if_on_one_line=true
```

**Reply by rgrunber** [reply]
You can also check out https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.jdt.doc.isv%2Freference%2Fapi%2Forg%2Feclipse%2Fjdt%2Fcore%2Fformatter%2FDefaultCodeFormatterConstants.html if you want to see a list of all possible format options. There's many.

**Comment by microwin7** [comment]
And another question, I can't find an alternative in org.eclipse.jdt.core.prefs
`"java.sources.organizeImports.starThreshold": 4,`

**Reply by rgrunber** [reply]
That's because it isn't an option that is controlled through that file. I think we define it and ensure we pass it everywhere we deal with configuring the import statements : https://github.com/eclipse/eclipse.jdt.ls/blob/26d5d9f3ba2581327f46adfcb6d9a1ee8ace2154/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/TypeProposalUtils.java#L168

We do the same thing for import order, and static import on-demand threshold.
